lib/mongo/server/monitor/connection.rb

```diff
@@ -131,7 +133,7 @@ def initialize(address, options = {})
         #
         # @since 2.0.0
         def timeout
-          @timeout ||= options[:connect_timeout] || CONNECT_TIMEOUT
+          @timeout ||= options[:connect_timeout]
```

Comment:
If I understand well this timeout is still used but not for connexion, does it make sense to keep it `connect_timeout` then? Especially if there's no default any more, it'll be infinite for most people, and even if they think about setting the `socket_timeout` to avoid any trouble, there'll still be an infinite timeout in the code, because the default specified in the doc is not applied.

I think this should either be `socket_timeout` or keep the default, WDYT @estolfo ?